src/controllers/trust.involved.controller.ts

```diff
@@ -0,0 +1,105 @@
+import { NextFunction, Request, Response } from 'express';
+import { TrusteeTypeChoice } from '../model/trustee.type.model';
+import * as config from '../config';
+import { logger } from '../utils/logger';
+
+const TRUST_INVOLVED_TEXTS = {
+  title: 'Individuals or entities involved in the trust',
+};
+
+const get = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): void => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    const templateName = config.TRUST_INVOLVED_PAGE;
+
+    return res.render(
+      templateName,
+      {
+        backLinkUrl: `${config.TRUST_DETAILS_URL}/${req.params['id']}`,
+        templateName,
+        pageParams: {
+          title: TRUST_INVOLVED_TEXTS.title,
+          checkYourAnswersUrl: config.CHECK_YOUR_ANSWERS_URL,
+        },
+      },
+    );
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+const post = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    if (req.body.noMoreToAdd) {
+      return res.redirect(config.CHECK_YOUR_ANSWERS_URL);
+    }
+
+    const typeOfTrustee = req.body.typeOfTrustee;
+    // the req.params['id'] is already validated in the has.trust.middleware but sonar can not recognise this.
+    const url = `${config.TRUST_INVOLVED_URL}/${req.params['id']}`;
+
+    switch (typeOfTrustee) {
+        case TrusteeTypeChoice.historical:
+          logger.info("TODO: Route to trust-historical-beneficial-owner page ");
+          if (isValidUrl (url) ) {
```

Comment:
are you able to validate the url just once before the switch?

Reply:
I could do but when we do the next stories then each URL will change so I thought it was easier to do it this way